container/go/cloudbuild.yaml

```diff
@@ -157,3 +157,35 @@ steps:
       - "gs://rules_docker/$COMMIT_SHA/loader-darwin-amd64"
     id: "push-loader-darwin-amd64"
     waitFor: ["build-darwin-amd64"]
+
+# Step: build the puller and loader release binaries for Darwin ARM64
+  - name: "l.gcr.io/google/bazel"
+    args:
+      - "build"
+      - "--platforms=@io_bazel_rules_go//go/toolchain:darwin_arm64"
+      - "//container/go/cmd/puller:puller"
+      - "//container/go/cmd/loader:loader"
+    id: "build-darwin-arm64"
+    waitFor: ["push-loader-darwin-amd64"]
```

Comment:
Why does arm64 build need to wait for amd64 steps?

Reply:
Was following the previous pattern of waiting for the previous push
https://github.com/bazelbuild/rules_docker/blob/def5c9644be8eddb2637c8619198152d3c2f0c33/container/go/cloudbuild.yaml#L129-L137

not sure whats right. Should darwin arm also wait for linux push? 🤷

Reply:
@gravypod Any ideas?

Reply:
Is this a blocker @linzhp?

Reply:
Not really, just feel inefficient.

Reply:
I don't disagree. Should we merge this and fix it later?